crates/pathfinder/src/rpc/api.rs

```diff
@@ -669,7 +656,18 @@ impl RpcApi {
                         _ => BlockStatus::AcceptedOnL2,
                     };
 
-                    Ok(TransactionReceipt::with_status(receipt, block_status))
+                    // We require the transaction so that we can return the right RPC type for the receipt.
+                    match StarknetTransactionsTable::get_transaction(&db_tx, transaction_hash)
+                        .context("Reading transaction from database")
+                        .map_err(internal_server_error)?
+                    {
+                        Some(transaction) => Ok(TransactionReceipt::with_status(
+                            receipt,
+                            block_status,
+                            &transaction,
+                        )),
+                        None => Err(ErrorCode::InvalidTransactionHash.into()),
+                    }
```

Comment:
This should probably get combined with the `StarknetTransactionsTable::get_receipt` call above. As in, get both transaction and receipt in one call.

Reply:
I can add an issue for such a refactor but would keep this out of the scope of this PR.

Reply:
Issue added: #445